### DISCUSSION TITLE
New Member Onboarding Resources

### DISCUSSION THREAD

**Discussion by aidnem** [team401/Docs#27] [General]
General discussion to post useful resources for getting up to speed. These should be added to the docs if we find them useful!

**Comment by aidnem** [comment]
[This video](https://www.youtube.com/watch?v=gYL-3Phk6F0) on how to use simulation and how to use the new 2025 wpilib units library is pretty good. It's quite long and most of the IO stuff is code that we've used last year but the demonstration of getting the robot to show up in 3d sim is useful. The demonstration of how to use the units library is also very useful for the 2025 season.

**Comment by jkleiber** [comment]
**Git Tutorial**
Interactive git tutorial (this was pretty fun ngl): https://learngitbranching.js.org/?locale=en_US

**Java Practice**
In the past when I also taught Java as part of doing programming mentoring I made the following "competitions" so students could learn Java skills:
1. [Java Skills Challenge](https://www.hackerrank.com/java-skills-challenge)
2. [Robotics I](https://www.hackerrank.com/rocket-robotics-1)
3. [Robotics II](https://www.hackerrank.com/robotics-ii)
4. [Robotics III](https://www.hackerrank.com/robotics-iii)

If you don't know much Java (or your skills are rusty), I would recommend trying some of these problems out to get more familiar. However, do not get hung up on these things too much - my mentoring philosophy has changed a lot since 2019 and I currently work on the belief that writing code that counts is the fastest way to learn.

**Intro to Justin**
It is valuable to understand my expectations and what I bring to the table as your mentor. Otherwise I'm just some mystery guy who is telling you what to do, and that's not very effective imo.

My background:
- I currently work as a GNC Engineer (basically a robotics engineer in aerospace) at Reliable Robotics, a company that creates unmanned aircraft systems like [this one](https://www.youtube.com/watch?v=w2O4hgNCMd4) ([raw video](https://www.youtube.com/watch?v=y2UNHvQj5rc)). I am the technical lead for the guidance system of this aircraft.
- My education: MS in computer engineering at Virginia Tech and BS in computer engineering at University of Oklahoma
- In high school I was on team 624, a relatively well known team from the Houston area, for the 2013-2016 FRC seasons

My expectations:
- I expect that students will work on programming outside of shop meetings. It's really hard to learn a new skill like programming by only practicing for a handful of hours per week. I operate using the "[flipped classroom](https://bokcenter.harvard.edu/flipped-classrooms)" technique, where shop time is best spent asking me questions and working together to make tactical decisions. Obviously we should still code some during shop (and use that precious time to test on hardware if needed), but most of the work should really be done when I'm not around - making mistakes on your own is the best way to learn how to code imo.
- That said, I expect that you will make mistakes. Everyone does. That is why we have code reviews and require verification evidence to merge code. Even then, those processes sometimes fail and we have to improve them. If I had to choose, I would want my students to work hard, make mistakes, and learn a lot rather than not make changes out of fear of being wrong.
- I expect that students will constantly strive to get a little bit better each day. Nobody is ever a "finished product" and we all can improve together. That expectation extends to wanting to make the team better (I want you to develop opinions about how the team can improve, and learn how to effect change when you see things that need to be fixed).
- I expect that you will help your fellow team members. A good engineer is someone who gets their own work done, but a great engineer is someone who does that plus enables other engineers to work efficiently. Some examples: writing great documentation, improving automated testing, making great examples for others to leverage, or writing a good code library for the team to use.

Looking forward to working and learning together!
